crates/proof-of-sql/src/base/database/record_batch_dataframe_conversion.rs

```diff
@@ -81,6 +83,42 @@ pub fn record_batch_to_dataframe(record_batch: RecordBatch) -> Option<DataFrame>
                         // Note: we make this unchecked because if record batch has values that overflow 38 digits, so should the data frame.
                         .into_series()
                 }
+                arrow::datatypes::DataType::Timestamp(time_unit, _timezone_option) => {
```

Comment:
NIT: it's probably a bit cleaner to do something like this:
```rust
arrow::datatypes::DataType::Timestamp(arrow::datatypes::TimeUnit::Second, _) => {
    let data = col
                                .as_any()
                                .downcast_ref::<arrow::array::TimestampSecondArray>()
                                .map(|array| array.values())
                                .unwrap();
                            Series::new(f.name(), data)
},
arrow::datatypes::DataType::Timestamp(arrow::datatypes::TimeUnit::Millisecond, _) => {
    let data = col
                                .as_any()
                                .downcast_ref::<arrow::array::TimestampMillisecondArray>()
                                .map(|array| array.values())
                                .unwrap();
                            Series::new(f.name(), data)
},
```
Sometimes nested matches are more efficient, but each branch is duplicated anyway, so in this case, unnesting them is probably a bit more concise.

You could do this in a bunch of different places in this PR.